variables.tf

```
@@ -179,3 +184,9 @@ variable "tags" {
   type        = map(string)
   description = "A mapping of tags to assign to resource"
 }
+
+variable "workspace_authentication_method" {
+  type        = string
+  default     = "user"
+  description = "User or role based authentication from the workspace to AWS"
+}
```

```suggestion
variable "workspace_authentication_method" {
  type        = string
  default     = "iam_user"
  description = "AWS account authentication method (can be iam_role or iam_user)"

  validation {
    condition     = lower(var.workspace_authentication_method) == "iam_role" || lower(var.workspace_authentication_method) == "iam_user"
    error_message = "The workspace_authentication_method value must be either \"iam_role\" or \"iam_user\"."
  }
}
```

Thought to prefix the valid options with `iam_` because they're planning to add OIDC based auth, and then that could be an `oidc_role`.